test/e2e/simple_vlan_and_dynamic_ip_test.go

```diff
@@ -0,0 +1,52 @@
+package e2e
+
+import (
+	"fmt"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("NodeNetworkState", func() {
+	Context("when vlan configured", func() {
+		var (
+			vlanId = "102"
+		)
+
+		BeforeEach(func() {
+			updateDesiredState(ifaceUpWithVlanUp(*firstSecondaryNic, vlanId))
+			waitForAvailableTestPolicy()
+		})
+		AfterEach(func() {
+			updateDesiredState(vlanAbsent(*firstSecondaryNic, vlanId))
+			resetDesiredStateForNodes()
+		})
+		It("should have the vlan interface configured", func() {
+			for _, node := range nodes {
+				interfacesNameForNodeEventually(node).Should(ContainElement(fmt.Sprintf(`%s.%s`, *firstSecondaryNic, vlanId)))
+				vlanForNodeInterfaceEventually(node, fmt.Sprintf(`%s.%s`, *firstSecondaryNic, vlanId)).Should(Equal(vlanId))
+			}
+		})
+	})
+	Context("when vlan and dynamic address configured", func() {
+		var (
+			vlanId = "102"
+		)
+		BeforeEach(func() {
+			updateDesiredState(ifaceUpWithVlanUp(*firstSecondaryNic, vlanId))
```

Comment:
Why do you need both this and the following line? We should just configure the VLAN in a single NNCP

Reply:
I can do it with one NNCP.
But the test is for then documentation and  in the doc I show how you can configure dynamic IP on top of existing vlan interface.

Reply:
Aha! That is useful. But please express it in the Context. "when dynamic address is configured on an existing VLAN interface". And then please, add `waitForAvailableTestPolicy()` call after this line, so we really reconfigure an existing interface